lib/trivial.nix

```diff
@@ -53,6 +53,41 @@ rec {
   # argument, but it's nice this way if several uses of `extends` are cascaded.
   extends = f: rattrs: self: let super = rattrs self; in super // f self super;
 
+  # Compose two extension functions:
+  #
+  #     nix-repl> h = self: super: { bar = "<${super.bar}>"; }
+  #
+  #     nix-repl> fix (extends (composeExtensions h g) f)
+  #     { bar = "<bar>"; foo = "foo + "; foobar = "foo + <bar>"; }
+  composeExtensions = f: g: self: super:
```

Comment:
No, that doesn't quite do what we want it to:

```
(extend f (extend g self)
#                   ^
#------------------/
```

That needs to be a function taking it's fixpoint (e.g. the `rattrs` naming convention), as that's what `extends` expects.